Search/Command/RebuildResumeManager.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/*
+ * This file is part of the MassiveSearchBundle
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Massive\Bundle\SearchBundle\Search\Command;
+
+/**
+ * Allow the storage of "checkpoints" which indicate how far
+ * a reindexing process has progressed in the case that it is
+ * interupted (f.e. by an error) with the aim of allowing the
+ * reindexing process to resume from where it left off.
+ *
+ * This is basically a key/value store.
+ */
+class RebuildResumeManager
+{
+    public function getCheckpointFile()
+    {
+        return sys_get_temp_dir() . '/' . str_replace('\\', '_', __CLASS__);
+    }
+
+    public function checkpoint($name, $value)
```

Comment:
It is called from the listener. It should be renamed to `setCheckpoint` or similar.